Credentials/credentials_student_deleteProcess.php

```diff
@@ -1,80 +1,66 @@
-if (isset($_GET['search'])) {
-    $search = $_GET['search'];
-}
-$allStudents = '';
-if (isset($_GET['allStudents'])) {
-    $allStudents = $_GET['allStudents'];
-}
-$credentialsCredentialID = $_GET['credentialsCredentialID'];
-$URL = $_SESSION[$guid]['absoluteURL'].'/index.php?q=/modules/'.getModuleName($_POST['address'])."/credentials_student_delete.php&gibbonPersonID=$gibbonPersonID&search=$search&allStudents=$allStudents&credentialsCredentialID=".$credentialsCredentialID;
-$URLDelete = $_SESSION[$guid]['absoluteURL'].'/index.php?q=/modules/'.getModuleName($_POST['address'])."/credentials_student.php&gibbonPersonID=$gibbonPersonID&search=$search&allStudents=$allStudents";
+$credentialsCredentialID = $_GET['credentialsCredentialID'] ?? '';
+
+$URL = $_SESSION[$guid]['absoluteURL'] . '/index.php?q=/modules/' . getModuleName($_POST['address']) . "/credentials_student_delete.php&gibbonPersonID=$gibbonPersonID&search=$search&allStudents=$allStudents&credentialsCredentialID=" . $credentialsCredentialID;
+$URLDelete = $_SESSION[$guid]['absoluteURL'] . '/index.php?q=/modules/' . getModuleName($_POST['address']) . "/credentials_student.php&gibbonPersonID=$gibbonPersonID&search=$search&allStudents=$allStudents";
 
 if (isActionAccessible($guid, $connection2, '/modules/Credentials/credentials_student_delete.php') == false) {
     //Fail 0
     $URL .= '&return=error0';
     header("Location: {$URL}");
 } else {
     //Proceed!
-    //Check if note specified
     if ($credentialsCredentialID == '' or $gibbonPersonID == '') {
-        echo 'Fatal error loading this page!';
+        echo __('Fatal error loading this page!');
     } else {
-        try {
-            $data = array('credentialsCredentialID' => $credentialsCredentialID);
-            $sql = 'SELECT * FROM credentialsCredential WHERE credentialsCredentialID=:credentialsCredentialID';
-            $result = $connection2->prepare($sql);
-            $result->execute($data);
-        } catch (PDOException $e) {
-            //Fail2
-            $URL .= '&return=error2';
-            header("Location: {$URL}");
-            exit();
-        }
+        $credentialGateway = $container->get(CredentialsCredentialGateway::class);
+        $credential = $credentialGateway->selectCredentialsCredentialById($credentialsCredentialID);
 
-        if ($result->rowCount() != 1) {
+        if (!$credential) {
             //Fail 2
             $URL .= '&return=error2';
             header("Location: {$URL}");
         } else {
             //Write to database
             try {
-                $data = array('credentialsCredentialID' => $credentialsCredentialID);
-                $sql = 'DELETE FROM credentialsCredential WHERE credentialsCredentialID=:credentialsCredentialID';
-                $result = $connection2->prepare($sql);
-                $result->execute($data);
-            } catch (PDOException $e) {
+                if (!$credentialGateway->deleteCredentialsCredential($credentialsCredentialID)) {
```

Comment:
If the `deleteCredentialsCredential` method returns false on failure, the try-catch could perhaps be simplified into an if-else 🤔